Decisions.AdobeSign/Utility/AdobeSignApiBase.cs

```diff
@@ -1,89 +1,90 @@
         {
-            CheckResponse(response);
+            if (!response.IsSuccessStatusCode)
+                throw new AdobeSignException(response.ReasonPhrase, response.StatusCode);
             var responseString = response.Content.ReadAsStringAsync().Result;
-
-            var result = JsonConvert.DeserializeObject<R>(responseString, jsonSettings);
-
-            return result;
+            return JsonConvert.DeserializeObject<TR>(responseString, JsonSettings);
         }
 
-
-        private static R GetRequest<R>(AdobeSignConnection connection, string requestUri) where R : new()
+        private static HttpResponseMessage SendAsync(HttpRequestMessage requestMessage)
         {
-            HttpResponseMessage response = GetJsonClient(connection).GetAsync(requestUri).Result;
-            return ParseResponse<R>(response);
+            HttpResponseMessage httpResponseMessage = HttpClients
+                .GetHttpClient(HttpClientAuthType.Normal)
+                .SendAsync(requestMessage).Result;
+            if (!httpResponseMessage.IsSuccessStatusCode)
+                throw new AdobeSignException(httpResponseMessage.ReasonPhrase, httpResponseMessage.StatusCode);
+            return httpResponseMessage;
         }
 
-        private static R PostRequest<R, T>(AdobeSignConnection connection, string requestUri, T content) where R : new()
+        private static HttpRequestMessage BuildHttpGetRequestMessage(
+            OAuthToken token,
+            string url,
+            string mediaType,
+            string contentType = null)
         {
-            string data = ParseRequestContent(content);
-            var contentStr = new StringContent(data, Encoding.UTF8, "application/json");
-
-            HttpResponseMessage response = GetJsonClient(connection).PostAsync(requestUri, contentStr).Result;
-            return ParseResponse<R>(response);
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token.TokenData);
+            requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(mediaType));
+            if (contentType != null)
+                requestMessage.Headers.TryAddWithoutValidation("Content-Type", contentType);
+            return requestMessage;
         }
-
+        
+        private static HttpRequestMessage BuildHttpPostRequestMessage(
+            OAuthToken token,
+            string url,
+            string contentType = null)
+        {
+            var requestMessage = new HttpRequestMessage(HttpMethod.Post, url);
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token.TokenData);
+            requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            if (contentType != null)
+                requestMessage.Headers.TryAddWithoutValidation("Content-Type", contentType);
```

Comment:
I think this will fail because HttpRequestMessage requires content headers be added on the Content property.